### PR TITLE
improve detection and messaging if parameters file is missing for both docdb & redis

### PR DESCRIPTION
**Pull request by sebastus**

### CODE REVIEW

File: lib/services/azurerediscache/index.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ Handlers.provision = function(log, params, next) {
   if (!cp.allValidatorsSucceed()) {
     var reply = Reply(HttpStatus.INTERNAL_SERVER_ERROR);
     reply.value.description = 'Input validators failed.';
-    var err = new Error('Validation failed.');
-    err.code = HttpStatus.INTERNAL_SERVER_ERROR;
+    var err = new Error('Parameter validation failed.  Did you supply a redis cache parameters file?');
+    err.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
```
Comment (bingosummer) [inline lib/services/azurerediscache/index.js]: Extra blank.

File: lib/services/azuredocdb/index.js [hunk 2]
```diff
@@ -30,8 +30,8 @@ Handlers.provision = function(log, params, next) {
   if (!cp.allValidatorsSucceed()) {
     var reply = Reply(HttpStatus.INTERNAL_SERVER_ERROR);
     reply.value.description = 'Input validators failed.';
-    var err = new Error('Validation failed.');
-    err.code = HttpStatus.INTERNAL_SERVER_ERROR;
+    var err = new Error('Parameter validation failed.  Did you supply a DocDb parameters file?');
```
Comment (bingosummer) [inline lib/services/azuredocdb/index.js]: Two blanks before `Did you supply a DocDb parameters file?`

File: lib/services/azurerediscache/cmd-provision.js [hunk 3]
```diff
@@ -117,7 +122,7 @@ var cacheProvision = function(log, params) {
             if (resourceGroupName.length !== 0) return true;
         }
 
-        log.error('Redis Cache Provision: Resource Group name was not provided.');
+        log.error('Redis Cache Provision: Resource Group name was not provided.  Did you supply the parameters file?');
```
Comment (bingosummer) [inline lib/services/azurerediscache/cmd-provision.js]: Two blanks before `Did you supply a DocDb parameters file?`

File: test/unit/services/azurerediscache/provision-spec.js [hunk 4]
```diff
@@ -60,6 +60,27 @@ describe('RedisCache - Provision - PreConditions', function() {
...
 
+describe('RedisCache - Provision - PreConditions incorrect', function() {
+    var validParams = {};
+    var cp;
+        
+    before(function() {
+        validParams = {  /* missing parameters!! */
+            instance_id : 'b259c5e0-7442-46bc-970c-9912613077dd',            
+            azure : azure,
+            provisioning_result: '{\"provisioningState\":\"Creating\"}'
```
Comment (bingosummer) [inline test/unit/services/azurerediscache/provision-spec.js]: I think, there should not be `provisioning_result` inside `validParams`.
It should be something like:
```
parameters: {
  location: 'southcentralus'
}
```
Other parameters are missing except `location`.

**Comment by bingosummer** [comment]
@sebastus 
I leave some comments on the codes.
Could you please rebase your commits into a single one?

File: lib/services/azurerediscache/index.js (hunk 1 above)
Comment (sebastus) [inline lib/services/azurerediscache/index.js]: Fixed.  But can you explain to me why this is important?  I also fixed a few more err.code instances in the same file.

File: lib/services/azuredocdb/index.js (hunk 2 above)
Comment (sebastus) [inline lib/services/azuredocdb/index.js]: In my high school days when I learned to type, they said "one blank after commas, two after periods".  Are they teaching something different these days?

File: lib/services/azurerediscache/cmd-provision.js (hunk 3 above)
Comment (sebastus) [inline lib/services/azurerediscache/cmd-provision.js]: Fixed.

File: test/unit/services/azurerediscache/provision-spec.js (hunk 4 above)
Comment (sebastus) [inline test/unit/services/azurerediscache/provision-spec.js]: You're correct - provisioningState shouldn't be part of the validParms for provision command.  In this particular case, I'm testing for missing parameters, so the location and everything else that is part of parameters object is missing intentionally.

**Comment by sebastus** [comment]
I'll squash once we're agreed that all issues are resolved.

File: lib/services/azurerediscache/index.js (hunk 1 above)
Comment (bingosummer) [inline lib/services/azurerediscache/index.js]: This is not important in the code logic. It's just a code convention, I think.

File: lib/services/azuredocdb/index.js (hunk 2 above)
Comment (bingosummer) [inline lib/services/azuredocdb/index.js]: Sorry, I didn't know "one blank after commas, two after periods" before. To me, I usually type one blank after periods, and never see two blanks in English articles.
I searched it, and found http://www.dailywritingtips.com/one-or-two-spaces-after-a-period/ for your reference.

Again, it's just a convention. You decide. :)

File: test/unit/services/azurerediscache/provision-spec.js (hunk 4 above)
Comment (bingosummer) [inline test/unit/services/azurerediscache/provision-spec.js]: Got it. I thought the case was for incomplete parameters. For example, there is only `location` in `parameters` and others (e.g. `sku`) are missing.
If the case is for missing parameters, that's fine. Thanks for explaining it.

**Comment by bingosummer** [comment]
LGTM. Thank you, @sebastus

File: lib/services/azuredocdb/index.js (hunk 2 above)
Comment (sebastus) [inline lib/services/azuredocdb/index.js]: I actually did learn to type on an actual typewriter.  Wasn't aware of the debate over the years.  Thanks for the reference.  Pretty funny stuff.  I'm good with the logic to use one space.  We'll see how good I am at changing my ways.

**Comment by bingosummer** [comment]
@sebastus 
Could you please squash into one commit? Then I can merge your PR.

**Comment by sebastus** [comment]
Yes, haven't forgotten.  Just busy busy busy!

**Comment by sebastus** [comment]
The changes you checked in to default debug messages caused conflicts with this PR.  I can't do the merge.  Can you work on it and let me know?  Then I'll squash.

**Comment by bingosummer** [comment]
@sebastus 
I resolved the conflict and pushed codes to a new branch `sebastus-master`. You can fetch it locally and compare it with your own branch using `git diff`. If `sebastus-master` is OK, you can close this PR and submit a new one. WDYT?

**Comment by sebastus** [comment]
used sebastus-master branch instead of this one.

**Pull request closed without merging**